### DISCUSSION TITLE
Gold gradient around text

### DISCUSSION THREAD

**Discussion by jaka1-newbie** [ImageMagick/ImageMagick#2795] [Help] [answered]
Have background.
Need to add 2 border.
One border with gold gradient like this.
![1](https://user-images.githubusercontent.com/67605008/98157991-e8140380-1ee2-11eb-9ec6-341b2ccb4c29.jpg)

**Comment by jaka1-newbie** [comment]
Please  help!

**Comment by fmw42** [comment]
Post your input image and show where you want the border. Is it around the outside? Or do you have a white region that you want the gold around.

**Comment by jaka1-newbie** [comment]
Any image.
Border like at start post picture.

**Comment by jaka1-newbie** [comment]
Border around Tranquil Rose Gold

**Comment by fmw42** [comment]
What version of Imagemagick and what platform?

**Comment by jaka1-newbie** [comment]
IM 7 
win 7 x64

**Comment by fmw42** [comment, marked as answer]
The basic idea is to make a diagonal gradient image. The create a smaller white label image. Then compose the white label image over the gradient. Then put a white border about that. Then create another larger diagonal gradient and compose the white image with gold border and white border over the second gradient image. Then composite that over the orginal background image.

![image-cat](https://user-images.githubusercontent.com/13458218/98301000-f576cf80-1f6e-11eb-9fe4-b67925347017.jpg)


```
magick image-cat.jpg ^
-set option:wd1 "%[fx:0.75*w]" -set option:ht1 "%[fx:0.75*h]" ^
-set option:wd2 "%[fx:0.75*w-2]" -set option:ht2 "%[fx:0.75*h-2]" ^
-set option:wd3 "%[fx:0.75*w+15]" -set option:ht3 "%[fx:0.75*h+15]" ^
( -size "%[wd1]x%[ht1]" xc: -sparse-color barycentric '0,0 rgb(155,125,50)  -%[wd1],%[ht1] rgb(155,125,50)  %[wd1],%[ht1] rgb(255,200,160)' ^
-size "%[wd2]x%[ht2]" xc:white -fill black -gravity center -pointsize 36 -annotate +0+0 "This Is A Cat" ^
-gravity center -compose over -composite ^
-bordercolor white -border 5 ^
( -size "%[wd3]x%[ht3]" xc: -sparse-color barycentric '0,0 rgb(155,125,50)  -%[wd3],%[ht3] rgb(155,125,50)  %[wd3],%[ht3] rgb(255,200,160)' ) ^
+swap -gravity center -compose over -composite ) ^
-gravity center -compose over -composite ^
image-cat_result.png
```

<br>


![image-cat_result](https://user-images.githubusercontent.com/13458218/98301034-058eaf00-1f6f-11eb-9a9e-0ff26054d47a.png)


For the diagonal gradient, see https://imagemagick.org/Usage/canvas/#diagonal_gradients

For the label image, see https://imagemagick.org/Usage/text/#annotate

For the -set artifacts, see https://imagemagick.org/Usage/basics/#define

**Comment by jaka1-newbie** [comment]
So cool!!! and very difficult(

**Comment by jaka1-newbie** [comment]
convert: invalid argument for option '-size': %[wd1]x%[ht1] @ error/convert.c/ConvertImageCommand/2816.

**Reply by fmw42** [reply]
There is no ":" after -size

**Comment by jaka1-newbie** [comment]
change magick to convert  because:
magick.exe: unable to open image 'magick': No such file or directory @ error/blob.c/OpenBlob/3539.
magick.exe: no decode delegate for this image format `' @ error/constitute.c/ReadImage/562.

**Comment by jaka1-newbie** [comment]
May be change size to points? Not %

**Reply by fmw42** [reply]
You cannot use -size in points in IM. -pointsize is in pixels 

On Windows you may need to double your % to %%, esp in .bat files

Do not change magick to convert. Try changing magick to magick.exe

**Comment by jaka1-newbie** [comment]
magick.exe: unable to open module file 'C:\Program Files\ImageMagick-7.0.10-Q16-HDRI\modules\coders\IM_MOD_RL_EXE_.dll': No such file or directory @ warning/module.c/GetMagickModulePath/675.
magick.exe: no decode delegate for this image format `EXE' @ error/constitute.c/ReadImage/562.

**Reply by fmw42** [reply]
What was your exact command?  Does this work?

`magick logo: logo.gif`

or does this work

`magick.exe logo: logo.gif`

If neither work, then you have not installed Imagemagick 7. Perhaps you have both IM 6 and IM 7 and are using the wrong one.

**Comment by jaka1-newbie** [comment]
Same problem
Reinstall IM

https://prnt.sc/veb8de
https://prnt.sc/veb8r6

Sure that point to folder with IM v7

**Comment by jaka1-newbie** [comment]
delete magick and work 

magick.exe: invalid argument for option 'sparse-color': Invalid number of Arguments @ error/operation.c/SparseColorOption/252.

**Reply by fmw42** [reply]
Show me your exact command.

**Comment by jaka1-newbie** [comment]
"D:\3.jpg" -set option:wd1 "%[fx:0.75*w]" -set option:ht1 "%[fx:0.75*h]" -set option:wd2 "%[fx:0.75*w-2]" -set option:ht2 "%[fx:0.75*h-2]" -set option:wd3 "%[fx:0.75*w+15]" -set option:ht3 "%[fx:0.75*h+15]" ( -size "%[wd1]x%[ht1]" xc: -sparse-color barycentric '0,0 rgb(155,125,50) -%[wd1],%[ht1] rgb(155,125,50)  %[wd1],%[ht1] rgb(255,200,160)' ) ( -size "%[wd2]x%[ht2]" xc:white -fill black -gravity center -pointsize 36 -annotate +0+0 "This Is A Cat" ) ( -clone 1,2 -gravity center -compose over -composite -bordercolor white -border 5 ) -delete 1,2 ( -size "%[wd3]x%[ht3]" xc: -sparse-color barycentric '0,0 rgb(155,125,50)  -%[wd3],%[ht3] rgb(155,125,50)  %[wd3],%[ht3] rgb(255,200,160)' ) ( -clone 1,2 +swap -gravity center -compose over -composite ) -delete 1,2 -gravity center -compose over -composite "D:\result1.jpg"

**Comment by jaka1-newbie** [comment]
change ' to " !!! ))) 

Got it !

**Reply by fmw42** [reply]
On Windows change single quotes to double quotes

**Comment by jaka1-newbie** [comment]
Can you send small investment?

**Reply by fmw42** [reply]
I do not understand?

**Reply by jaka1-newbie** [reply]
Some $ for help

**Comment by jaka1-newbie** [comment]
How can i set coordinates of this border?

**Reply by fmw42** [reply]
add -geometry

-gravity center -geometry +X+Y -compose over -composite "D:\result1.jpg"

Set +-X and +-Y relative to the center.

Or 

-gravity northwest -geometry +X+Y -compose over -composite "D:\result1.jpg"

Set +-X and +-Y relative to the top left corner.

**Comment by jaka1-newbie** [comment]
And can we do more complex gradient?

**Reply by fmw42** [reply]
Please explain and show the kind of gradient that you want?

**Reply by jaka1-newbie** [reply]
![Uploading 1593745203_51-p-foni-s-bezhevim-gradientom-59.jpg…]()

**Reply by fmw42** [reply]
Your file is not accessible.

**Comment by jaka1-newbie** [comment]
![1593745203_51-p-foni-s-bezhevim-gradientom-59](https://user-images.githubusercontent.com/67605008/98334173-f8270280-200a-11eb-99f3-47f1f03b7e1a.jpg)
![golden-cone-gradient-vector-21113068](https://user-images.githubusercontent.com/67605008/98334178-f9f0c600-200a-11eb-9d9f-2bcc104dfb3b.jpg)

**Comment by fmw42** [comment]
Here are two gradients:

```
magick -size 500x500 \
-define gradient:direction=southeast \
gradient:"rgb(189,137,44)-rgb(251,231,187)" \
-implode 1 -swirl 360 \
grad1.png
```

![grad1](https://user-images.githubusercontent.com/13458218/98395811-a59d1480-2011-11eb-8310-54466830ec55.png)


```
magick -size 500x500 \
-define gradient:direction=east gradient:"black-white" \
-solarize 50% -level 0x50% \
+level-colors "rgb(189,137,44),rgb(251,231,187)" \
-implode 1 -swirl 360 -blur 0x15 \
grad2.png
```

![grad2](https://user-images.githubusercontent.com/13458218/98395838-ad5cb900-2011-11eb-87b0-37b384259b3f.png)


See https://imagemagick.org/Usage/canvas/ for numerous gradients and patterns

Also see https://www.imagemagick.org/script/gradient.php
